### PR TITLE
Tab bar styling

### PR DESCRIPTION
**Pull request by conrad-vanl**
- [x] Flush out styling for Tab Bar.
- [x] Adds in a component to use temporarily for dealing with iPhone X until we upgrade to React Native 50.x
- [x] Handle link active states better. Currently not working
- [ ] Implement Typography components (probably out of scope for this branch)

## Screenshots

image

image

image

## Creator

- [ ] Build relevant tests if any apply
- [ ] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [ ] Set a relevant reviewer

## Reviewer

- [x] Run `test` and make sure all tests pass
- [x] Review function and form on iOS
- [x] Review function and form on Android
- [x] Review function and form on Web
- [ ] Review new test logic

### CODE REVIEW

**Comment by mary-pr-poppins** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  No Changelog changes!
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/@modules/TabBar/Link.js [hunk 1]
```diff
@@ -1,58 +1,76 @@
...
-    color: 'white',
+  largeButton: {
+    height: 80,
   },
 });
 
-class FooterNavLink extends PureComponent {
-  static propTypes = {
+// Styled View to wrap icon and label
+// TODO: should "styled" components like this always go in their own file?
```
Comment (burkeshartsis) [inline src/@modules/TabBar/Link.js]: >// TODO: should "styled" components like this always go in their own file?

I would vote yes.

File: src/@modules/TabBar/Link.js [hunk 2]
```diff
@@ -1,58 +1,76 @@
...
+  mapProps(({
+    match, location, history, ...ownProps
+  }) => ({
+    isActive: matchLocationToPath(ownProps.to, location),
+    ...ownProps,
+  })),
+);
+
+// Get color from active route
+const determineColorFromActiveRoute =
```
Comment (burkeshartsis) [inline src/@modules/TabBar/Link.js]: Why do you feel the need to move this out of the `compose()` function below on line 50? I get that on line 34 above you're consolidating HOC's a bit but are we not going to have to do this a lot? Is this a pattern we want to continue.

File: src/@modules/TabBar/Link.js [hunk 3]
```diff
@@ -1,58 +1,76 @@
...
+const TabBarLink = enhance(({
+  label = null,
+  icon = null,
+  color = null,
+  ...linkProps
+}) => (
+  <Link {...linkProps}>
+    <LinkContainer>
+      {icon ? <Icon name={icon} fill={color} /> : null}
+      <MediaQuery minWidth="md">{label ? <Text style={{ color }}>{label}</Text> : null}</MediaQuery>
```
Comment (burkeshartsis) [inline src/@modules/TabBar/Link.js]: While I don't think render performance matters here based how I understand the navigations components to work there just wont be that many prop changes. That said, if we don't build things to be pure from the beginning a mistake somewhere else could cause unexpected rerenders here. To fix this, all you have to do is move the style literal `{{ color }}` outside of the return. Maybe call it `labelColor`.

File: src/@modules/TabBar/index.js [hunk 4]
```diff
@@ -11,16 +13,19 @@ const styles = StyleSheet.create({
...
+    paddingHorizontal: 10,
   },
 });
 
 export default compose(
   withTheme(({ theme: { darkPrimaryColor } = {} }) => ({ darkPrimaryColor })),
   styled(({ darkPrimaryColor }) => ({ backgroundColor: darkPrimaryColor })),
   mediaQuery(({ md }) => ({ maxWidth: md }), styled(styles.horizontal), styled(styles.vertical)),
-)(View);
+  mapProps(props => omit(props, ['darkPrimaryColor'])),
```
Comment (burkeshartsis) [inline src/@modules/TabBar/index.js]: Just curious, why the need to omit this prop?

File: src/@modules/TabBar/Link.js (hunk 2 above)
Comment (conrad-vanl) [reply src/@modules/TabBar/Link.js]: Trying to make it easier to read

**Pull request merged**